### PR TITLE
Add --only-rerun flag to add option to rerun specific errors

### PR DESCRIPTION
**Pull request by gnikonorov**
Add new flag `--only-rerun` to allow for users to only rerun errors matching regexes passed to this flag. If the flag is omitted or passed in without a value then no changes occur and all failures are rerun.

Closes https://github.com/pytest-dev/pytest-rerunfailures/issues/101

### CODE REVIEW

File: pytest_rerunfailures.py [hunk 1]
```diff
@@ -30,6 +31,14 @@ def pytest_addoption(parser):
     group = parser.getgroup(
         "rerunfailures",
         "re-run failing tests to eliminate flaky failures")
+    group._addoption(
+        '--only-rerun',
+        action='store',
+        dest='only_rerun',
+        type=str,
+        default=None,
+        help='Optional comma separated list. If passed, only rerun errors matching the regexes provided.'
```
Comment (icemac) [inline pytest_rerunfailures.py:40]: I was wondering if there is the ability to use this option multiple times, so we get a list of regexes from the parser. This would allow commas in the regex and the shell eliminates not necessary whitespace for us.

Comment (gnikonorov) [reply pytest_rerunfailures.py:40]: Let me check tonight ( I’m in EST )

Comment (gnikonorov) [reply pytest_rerunfailures.py]: setting action=append for `group._addoption` does that. Thank you for bringing this up, I updated the PR accordingly

**Comment by gnikonorov** [comment]
I updated this PR according to your feedback @icemac

File: test_pytest_rerunfailures.py [hunk 2]
```diff
@@ -409,27 +409,31 @@ def pytest_runtest_logfinish(nodeid, location):
...
+            ('def test_only_rerun(): raise AssertionError("ERR")', ['Assertion*'], True),
+            ('def test_only_rerun(): raise AssertionError("ERR")', ['Assertion'], True),
+            ('def test_only_rerun(): raise AssertionError("ERR")', ['ValueError'], False),
+            ('def test_only_rerun(): raise AssertionError("ERR")', [''], True),
+            ('def test_only_rerun(): raise AssertionError("ERR")', ['AssertionError: '], True),
+            ('def test_only_rerun(): raise AssertionError("ERR")', ['AssertionError: ERR'], True),
+            ('def test_only_rerun(): raise AssertionError("ERR")', ['ERR'], True),
+            ('def test_only_rerun(): raise AssertionError("ERR")', ['AssertionError,ValueError'], False),
+            ('def test_only_rerun(): raise AssertionError("ERR")', ['AssertionError ValueError'], False),
+            ('def test_only_rerun(): raise AssertionError("ERR")', ['AssertionError','ValueError'], True),
```
Comment (icemac) [inline test_pytest_rerunfailures.py:424]:
```suggestion
            ('def test_only_rerun(): raise AssertionError("ERR")', ['AssertionError', 'ValueError'], True),
```

**Comment by gnikonorov** [comment]
Thank you. I'll merge this, and open a a PR for the README and changelog.

There's no rush for a release, thank you for asking.

**Comment by gnikonorov** [comment]
Added the README and changelog changes here, since it doesn't make sense to do otherwise. Thanks for reminding me @icemac. Please let me know if you're still comfortable with me merging these changes

File: pytest_rerunfailures.py [hunk 3]
```diff
@@ -173,6 +182,21 @@ def _remove_failed_setup_state_from_session(item):
     setup_state.stack = list()
 
 
+def _should_hard_fail_on_error(session_config, report):
+    if report.outcome != 'failed':
```
Comment (sallner) [inline pytest_rerunfailures.py:186]: It seems to me, that this test is also done in line 230 with `report.failed` or am I missing some edge case here? Which test case would fail, if we leave it out?

Comment (gnikonorov) [reply pytest_rerunfailures.py:186]: This is unrelated since it says that anything that is not failed cannot be a hard failure. Which makes sense to me, since otherwise this function would potentially flag non failed things as hard failures

Comment (sallner) [reply pytest_rerunfailures.py:186]: I understand the reason but I do not see, why we need it twice currently. It seems to be a guard for a case, that is not possible at the moment. But I would like to refactor line 230 anyhow, as this is not so easy to comprehend any more.

Comment (gnikonorov) [reply pytest_rerunfailures.py:186]: I’m working on [this issue](https://github.com/pytest-dev/pytest-rerunfailures/issues/43) and in the change I have a refactor of that line.

This change came out of it. It’s just unrelated, so I raised it first.

**Comment by graingert** [comment]
@S-Borkowski https://github.com/pytest-dev/pytest-rerunfailures/compare/9.0...master

**Comment by S-Borkowski** [comment]
When I run it like this: 
```
pytest -n auto --dist loadscope --reruns 3 --only-rerun PreSubmitError pathToTest
```
I get and error: pytest: error: unrecognised arguments: --only-rerun pathToTest

**Comment by S-Borkowski** [comment]
Downloaded version 9.0 (most recent) from pypi. I am unable to install packages through console; i need .whl file to install

**Comment by The-Compiler** [comment]
@S-Borkowski 9.0 was released on March 18th, this PR was merged on July 17th. You'll need to either find a way to install from the git repository, or wait until the next release.

**Comment by S-Borkowski** [comment]
@The-Compiler When will approx. be the next release? Is there any way You could provide me with .whl of July's version?

**Comment by The-Compiler** [comment]
@S-Borkowski  Here's the result of running `python3 setup.py bdist_wheel` in this repo: 
[pytest_rerunfailures-9.1.dev0-py3-none-any.zip](https://github.com/pytest-dev/pytest-rerunfailures/files/5119984/pytest_rerunfailures-9.1.dev0-py3-none-any.zip) - rename it from `.zip` to `.whl` to use it. I haven't tested it in any way, so use it on your own risk.

No idea when a new release is planned - that's up to the maintainers to decide.

**Comment by sallner** [comment]
@S-Borkowski @The-Compiler I had setup a milestone with some Tickets to be solved before a 9.1 release. In case you definitely need it quickly, I could cut a release tomorrow and move the remaining issues to a 9.2 release.

**Comment by S-Borkowski** [comment]
@sallner If it won't mess with Your work flow much, I would love that!

**Comment by S-Borkowski** [comment]
@sallner Please let me know what was your decision, and when i can expect 9.1 release :)

**Comment by sallner** [comment]
@S-Borkowski @The-Compiler  I have just released version 9.1 of this package. Feel free to use it. https://pypi.org/project/pytest-rerunfailures/9.1/

**Comment by S-Borkowski** [comment]
@sallner <3
